android/java/org/chromium/chrome/browser/customtabs/features/partialcustomtab/BravePartialCustomTabBottomSheetStrategy.java

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.customtabs.features.partialcustomtab;
+
+import android.app.Activity;
+
+import org.chromium.base.supplier.Supplier;
+import org.chromium.chrome.browser.browserservices.intents.BrowserServicesIntentDataProvider;
+import org.chromium.chrome.browser.fullscreen.FullscreenManager;
+import org.chromium.chrome.browser.lifecycle.ActivityLifecycleDispatcher;
+import org.chromium.chrome.browser.tab.Tab;
+import org.chromium.components.browser_ui.widget.TouchEventProvider;
+
+public class BravePartialCustomTabBottomSheetStrategy extends PartialCustomTabBottomSheetStrategy {
+
+    public BravePartialCustomTabBottomSheetStrategy(
+            Activity activity,
+            BrowserServicesIntentDataProvider intentData,
+            Supplier<TouchEventProvider> touchEventProvider,
+            Supplier<Tab> tab,
+            OnResizedCallback onResizedCallback,
+            OnActivityLayoutCallback onActivityLayoutCallback,
+            ActivityLifecycleDispatcher lifecycleDispatcher,
+            FullscreenManager fullscreenManager,
+            boolean isTablet,
+            boolean startMaximized,
+            PartialCustomTabHandleStrategyFactory handleStrategyFactory) {
+        super(
+                activity,
+                intentData,
+                touchEventProvider,
+                tab,
+                onResizedCallback,
+                onActivityLayoutCallback,
+                lifecycleDispatcher,
+                fullscreenManager,
+                isTablet,
+                startMaximized,
+                handleStrategyFactory);
+    }
+
+    public boolean mStopShowingSpinner;
+
+    @Override
+    public void onDragStart(int y) {
+        super.onDragStart(y);
+        mStopShowingSpinner = true;
```

Comment:
let's do that for our tab only, just make a detection like you did for a page refresh for the schema and chat URL. We better don't break other pages inside custom tabs.

Reply:
Sorry, missed this point though we had discussed.
Fixed at https://github.com/brave/brave-core/pull/21927/commits/ba776f2dacb6a6a601c5f5a48508ae16c9557718 .